editor/editor_properties_array_dict.h

```diff
@@ -174,22 +172,36 @@ class EditorPropertyDictionary : public EditorProperty {
 	Button *button_add_item = nullptr;
 	EditorPaginator *paginator = nullptr;
 	PropertyHint property_hint;
+	Variant::Type key_type;
+	PropertyHint key_type_hint;
+	String key_type_hint_string;
+	Variant::Type value_type;
+	PropertyHint value_type_hint;
+	String value_type_hint_string;
 
-	void _page_changed(int p_page);
-	void _edit_pressed();
-	void _property_changed(const String &p_property, Variant p_value, const String &p_name = "", bool p_changing = false);
-	void _change_type(Object *p_button, int p_index);
-	void _change_type_menu(int p_index);
+	void initialize_dictionary(Variant &p_dictionary);
 
-	void _add_key_value();
-	void _object_id_selected(const StringName &p_property, ObjectID p_id);
+	void _page_changed(int p_page);
 
 protected:
+	Ref<EditorPropertyDictionaryObject> object;
+
+	bool updating = false;
+
 	static void _bind_methods();
 	void _notification(int p_what);
 
+	virtual void _edit_pressed();
+	virtual void _property_changed(const String &p_property, Variant p_value, const String &p_name = "", bool p_changing = false);
+	virtual void _change_type(Object *p_button, int p_index);
+	virtual void _change_type_menu(int p_index);
+
+	virtual void _add_key_value();
+	virtual void _object_id_selected(const StringName &p_property, ObjectID p_id);
+	virtual void _remove_pressed(int p_index);
+
```

Comment:
why are all of those modifications needed?

Reply:
If you're talking about moving these functions from `private` to `protected`, then it's not strictly needed. Instead, it serves to mirror the current access levels of `EditorPropertyArray`. With how much else went into mirroring functionality & implementation, it wasn't too farfetched to adjust this as well

Reply:
most likely in this case it should be the other way around (ie. dictionary is right and it should be changed in Array) as it does not seem to have any reasons to be protected instead of private but also to be virtual I think.

Reply:
It's that way for array because the class is inherited by `EditorPropertyFontNamesArray`; now dictionary is setup to allow for similar inheritance if needed in the future

Reply:
So checked a bit more and it is right for array but should not be replicated in dictionary as there is no reasons to as there is no class inheriting from it (array as one which overrides those functions I assume) I would keep them as private.

Reply:
Oh I see now but then I think there is no reasons to bring that to EditorPropertyDictionary

Reply:
In terms of immediate use, it's true there's no real reason for it. However, as it doesn't end up making anything publically accessible that wasn't before, there isn't any real harm to it either. As there's already other function additions/changes in the class, I'm leaving this as-is